backend/src/user/hashUserPassword.ts

```diff
@@ -0,0 +1,10 @@
+import bcrypt from "bcrypt";
+import { UserInput } from "../graphql/resolvers-types";
+
+export const hashUserPassword = async (user: UserInput): Promise<UserInput> => {
+  if (user.password) {
+    const hashedPassword = await bcrypt.hash(user.password, 10);
```

Comment:
This could benefit from some error handling maybe? 

We could also make 10 a bit more readable: 

```
// more rounds means slower hashing but higher security
const saltRounds = 10;   
```